pkgs/applications/editors/emacs/generic.nix

```diff
@@ -138,7 +138,13 @@ let emacs = stdenv.mkDerivation (lib.optionalAttrs nativeComp {
     ++ lib.optional withXwidgets "--with-xwidgets"
     ++ lib.optional nativeComp "--with-native-compilation"
     ++ lib.optional withImageMagick "--with-imagemagick"
-    ;
+  ;
+
+  # Emacs checks for arm-apple-darwin, but our configuration is
+  # aarch64-apple-darwin. Force enable DO_CODESIGN.
+  preConfigure = lib.optionalString (stdenv.isDarwin && stdenv.isAarch64) ''
```

Comment:
I looked at this in more detail, and it looks like emacs itself has been churning between `arm` and `aarch64`, but the [most recent commit](https://git.savannah.gnu.org/cgit/emacs.git/commit/?id=a88f63500e475f842e5fbdd9abba4ce122cdb082) uses `aarch64`. If we backport this change as a patch then we know when to remove it (when it conflicts).

Reply:
Ok, should we change the comment, or keep it like this?

Reply:
Hmm. Add a link to the upstream change for context. This `sed` is probably the easiest way to make the required change.

Reply:
> Add a link to the upstream change for context.

^ I think this suggestion is still open, @hlolli. 🙂

Looking forward to these changes getting merged.

Reply:
Thanks for reminding me! I misunderstood as well. This part I copy pasted from the @thefloweringash's previous wip PR. Anyway, late for me here, tomorrow!

Reply:
ok I pushed a fetchpatch call to the svannah commit. Which should error when the patch doesn't apply anymore!